src/core/commands/start.ts

```diff
@@ -1,21 +1,42 @@
+import { eq } from "drizzle-orm";
 import { db } from "../db";
-import { intents } from "../db/schema";
+import { type Intent, intents } from "../db/schema";
+import { findActiveIntent } from "./findActiveIntent";
 
-export function start({
+export async function start({
   message,
   branchId,
 }: {
   message: string;
   branchId: string;
-}) {
-  const result = db
-    .insert(intents)
-    .values({
-      message,
-      status: "active",
-      branchId,
-    })
-    .run();
+}): Promise<Intent> {
+  try {
+    const activeIntent = await findActiveIntent({ branchId });
 
-  return Number(result.lastInsertRowid);
+    return db.transaction((tx) => {
+      if (activeIntent) {
+        db.update(intents)
+          .set({
+            status: "completed",
```

Comment:
It restarted before it finished, so ‘cancelled’ feels more accurate than ‘completed.’ What do you think?

Reply:
I feel like that `completed` is more like a temporary value as we haven't added a proper way to finish/cancel the active intent. I think the correct behavior would be to prompt the user to ask if they want to mark it as completed or cancelled, but yes let's set it to `cancelled` for now.